src/server/services/training.service.ts

```diff
@@ -36,50 +38,32 @@ type TrainingRequest = {
   fileMetadata: FileMetadata | null;
 };
 
-type moveAssetResponse = {
-  found?: boolean;
-  fileSize?: number;
-};
-
 const assetUrlRegex =
   /\/v\d\/consumer\/jobs\/(?<jobId>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\/assets\/(?<assetName>\S+)$/i;
 
 export const moveAsset = async ({ url, modelId }: MoveAssetInput) => {
-  if (!env.GENERATION_ENDPOINT) throw throwBadRequestError('Missing GENERATION_ENDPOINT env');
-  if (!env.ORCHESTRATOR_TOKEN) throw throwBadRequestError('Missing ORCHESTRATOR_TOKEN env');
-
   const urlMatch = url.match(assetUrlRegex);
   if (!urlMatch || !urlMatch.groups) throw throwBadRequestError('Invalid URL');
   const { jobId, assetName } = urlMatch.groups;
 
   const { url: destinationUri } = await getPutUrl(`model/${modelId}/${assetName}`);
 
-  const reqBody = {
-    $type: 'copyAsset',
+  const reqBody: Orchestrator.Training.CopyAssetJobPayload = {
     jobId,
     assetName,
     destinationUri,
   };
 
-  const response = await fetch(`${env.GENERATION_ENDPOINT}/v1/consumer/jobs?wait=true`, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json',
-      Authorization: `Bearer ${env.ORCHESTRATOR_TOKEN}`,
-    },
-    body: JSON.stringify(reqBody),
-  });
-
+  const response = await orchestratorCaller.copyAsset({ payload: reqBody });
```

Comment:
@flipperbw can you review this whole file? I made some improvements and I think you might wanna take a look